addons/mail/static/src/core/messaging_service.js

```diff
@@ -205,6 +205,32 @@ export class Messaging {
                         );
                     }
                     break;
+                case "discuss.channel/delete":
+                    {
+                        const thread = this.threadService.insert({
+                            id: notif.payload.id,
+                            model: "discuss.channel",
+                        });
+                        this.store.discuss.starred.messages =
+                            this.store.discuss.starred.messages.filter(
+                                (message) =>
+                                    message.resId !== thread.id && message.resModel !== thread.model
+                            );
+                        this.store.discuss.inbox.messages =
+                            this.store.discuss.inbox.messages.filter(
+                                (message) =>
+                                    message.resId !== thread.id && message.resModel !== thread.model
+                            );
+                        this.store.discuss.inbox.counter -= thread.message_needaction_counter;
+                        thread.messages.forEach((message) => {
+                            delete this.store.messages[message.id];
+                        });
```

Comment:
In general we prefer `for of` loops instead of `.forEach()`, because usually we expect method calls on array to have no side-effects. With `forEach` here, it gives the wrong impression at a glance there's no side-effects.

```js
for (const message of thread.messages) {
   delete this.store.messages[message.id];
}
```